src/stcal/resample/resample.py

```diff
@@ -0,0 +1,1530 @@
+                "var_flat",
+                model=model,
+                **pars,
+            )
+            if flat_var is not None:
+                mask = (flat_var >= 0) & np.isfinite(flat_var) & (weight > 0)
+                self._var_flat_wsum[mask] = np.nansum(
+                    [
+                        self._var_flat_wsum[mask],
+                        flat_var[mask] * weight[mask] * weight[mask]
+                    ],
+                    axis=0
+                )
+                self._var_flat_weight[mask] += weight[mask]
+
+    def finalize_resample_variance(self, output_model, free_memory=True):
+        """ Compute variance for the resampled image from running sums and
+        weights. Free memory (when ``free_memory=True``) that holds these
+        running sums and weights arrays.
+        """
+        # Divide by the total weights, squared, and set in the output model.
+        # Zero weight and missing values are NaN in the output.
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore", "invalid value*", RuntimeWarning)
+            warnings.filterwarnings("ignore", "divide by zero*", RuntimeWarning)
+
+            output_variance = (
+                self._var_rnoise_wsum / self._var_rnoise_weight /
+                self._var_rnoise_weight
+            ).astype(
+                dtype=self.output_array_types["var_rnoise"]
+            )
+            output_model["var_rnoise"] = output_variance
+
+            output_variance = (
+                self._var_poisson_wsum / self._var_poisson_weight /
+                self._var_poisson_weight
+            ).astype(
+                dtype=self.output_array_types["var_poisson"]
+            )
+            output_model["var_poisson"] = output_variance
+
+            output_variance = (
+                self._var_flat_wsum / self._var_flat_weight /
+                self._var_flat_weight
+            ).astype(
+                dtype=self.output_array_types["var_flat"]
+            )
+            output_model["var_flat"] = output_variance
+
+        if free_memory:
+            self._finalized = True
+            del (
+                self._var_rnoise_wsum,
+                self._var_poisson_wsum,
+                self._var_flat_wsum,
+                self._var_rnoise_weight,
+                self._var_poisson_weight,
+                self._var_flat_weight,
+            )
+
+    def _resample_one_variance_array(self, name, model, iscale,
+                                     weight_map, pixmap,
+                                     xmin=None, xmax=None, ymin=None,
+                                     ymax=None):
+        """Resample one variance image from an input model.
+
+        The error image is passed to drizzle instead of the variance, to
+        better match kernel overlap and user weights to the data, in the
+        pixel averaging process. The drizzled error image is squared before
+        returning.
```

Comment:
```suggestion
        The error image is passed to drizzle instead of the variance in order to
        better match kernel overlap and user weights to the data during the
        pixel averaging process. The drizzled error image is squared before
        returning.
```